lib/jekyll/frontmatter_defaults.rb

```diff
@@ -97,21 +97,26 @@ def applies?(scope, path, type)
       applies_path?(scope, path) && applies_type?(scope, type)
     end
 
+    # rubocop:disable Metrics/AbcSize
     def applies_path?(scope, path)
       return true if !scope.key?("path") || scope["path"].empty?
 
       sanitized_path = Pathname.new(sanitize_path(path))
-
-      site_path = Pathname.new(@site.source)
+      site_path      = Pathname.new(@site.source)
       rel_scope_path = Pathname.new(scope["path"])
       abs_scope_path = File.join(@site.source, rel_scope_path)
-      Dir.glob(abs_scope_path).each do |scope_path|
-        scope_path = Pathname.new(scope_path).relative_path_from site_path
-        return true if path_is_subpath?(sanitized_path, scope_path)
+
+      if glob_pattern?(rel_scope_path)
```

Comment:
So instead of this, we do `if scope["path"].to_s.include?("*")`. Or you can change the implementation of `glob_pattern?` to just do `path.to_s.include?("*")`. What do you think?